### ISSUE TITLE
Session timeout while editing content

### ISSUE THREAD

**Issue by cebe** [luyadev/luya-module-admin#128]
I just faced the situation that while typing a long text in CMS the session timed out and I got logged out automatically while typing. There should not be an auto login while a user is actively working in the backend.

**Comment by nadar** [comment]
I am sorry to hear that. Where you editing inside a textarea for the whole time? Because if you trigger any action (click somehwere, save, update, drag) this will reset the timeout timer to 0.

**Comment by cebe** [comment]
Yes, this was one text area open and typing all the time.

**Comment by cebe** [comment]
Is there a setting for the timout? Or how is it determined?

**Comment by cebe** [comment]
> Maybe check the last timestamp of a key stroke and provide this information to the timestamp ticker which is running each 2 min.

could be an option.

**Comment by cebe** [comment]
It may also be useful to show the time to logout somewhere, to make users aware that it exists. My bank for example has this:

![bildschirmfoto von 2018-05-22 23-15-48](https://user-images.githubusercontent.com/189796/40390635-2797563c-5e16-11e8-976a-5c5a56e19b7b.png)

**Comment by nadar** [comment]
Good idea, will do some research tomorrow and provide a good solution! Thanks and sorry.

**Comment by cebe** [comment]
Thanks for the fast response! I really like what you have created with luya so far :)

**Comment by nadar** [comment]
@TheMaaarc Maybe we could use the circle of the users online icon in order to indicate the idle time.

red/black = percentage idling
green = percentage until logout

(or the opposite around, not sure yet what makes more sense in terms of color)

<img width="81" alt="bildschirmfoto 2018-05-22 um 23 29 22" src="https://user-images.githubusercontent.com/3417221/40391309-148e541c-5e18-11e8-890e-0e5d357a991b.png">

(something like this - well yes could not find any better picture by googling with keyword "circle outline border indicates green")
![image](https://user-images.githubusercontent.com/3417221/40391423-73f0ed5c-5e18-11e8-978e-2d90847ef087.png)

**Comment by nadar** [comment]
@cebe give it a try if you like, let me know what you think. Thanks for the input and sorry for the data loss.

**Comment by cebe** [comment]
nice :+1: thanks!
